src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Fixer.cs

```diff
@@ -0,0 +1,199 @@
+            List<SyntaxNode> newArguments = new List<SyntaxNode>();
+            bool shouldTokenUseName = false;
+            string paramName = string.Empty;
+
+            // In C#, invocation.Arguments contains the arguments in the order passed by the user
+            // In VB, invocation.Arguments contains the arguments in the official parameter order
+            for (int i = 0; i < invocation.Arguments.Length; i++)
+            {
+                IArgumentOperation argument = invocation.Arguments[i];
+
+                // The type name is detected even if using an alias
+                if (!argument.Parameter.Type.Name.Equals("CancellationToken", StringComparison.Ordinal))
+                {
+                    if (!argument.IsImplicit)
+                    {
+                        SyntaxNode newArg;
+                        if (IsArgumentNamed(argument))
+                        {
+                            newArg = generator.Argument(argument.Parameter.Name, argument.Parameter.RefKind, argument.Value.Syntax);
+                            shouldTokenUseName = true;
+                        }
+                        else
+                        {
+                            newArg = argument.Syntax;
+                        }
+                        newArguments.Add(newArg);
+                    }
+                    else
+                    {
+                        shouldTokenUseName = true;
+                    }
+                }
+                else
+                {
+                    // Only reachable if the current method is the one that contains the ct
+                    // Won't be reached if it's an overload that contains the paramName
+                    paramName = argument.Parameter.Name;
+                }
+            }
+
+            // Create and append new ct argument to pass to the invocation, using the ancestor method parameter name
+            SyntaxNode cancellationTokenIdentifier = generator.IdentifierName(cancellationTokenParameterName);
+            SyntaxNode cancellationTokenNode;
+            if (shouldTokenUseName)
+            {
+                // If the paramName is unknown at this point, it's because an overload contains the ct parameter
+                // and since it cannot be obtained, no fix will be provided or else CA8323 shows up:
+                // CA8323: Named argument 'argName' is used out-of-position but is followed by an unnamed argument
+                if (string.IsNullOrEmpty(paramName))
+                {
+                    return false;
+                }
+
+                cancellationTokenNode = generator.Argument(paramName, RefKind.None, cancellationTokenIdentifier);
+            }
+            else
+            {
+                cancellationTokenNode = generator.Argument(cancellationTokenIdentifier);
+            }
+            newArguments.Add(cancellationTokenNode);
+
+            SyntaxNode newInvocation;
```

Comment:
I might be too much fond of switch pattern but that also looks like a good fit.

Reply:
Changed.